python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -208,6 +213,81 @@ async def prepare_for_flat(self, config_name: str) -> None:
             task_setup_electrometer,
         )
 
+    async def calculate_optimized_exposure_times(
+        self, wavelengths: list, config_data: dict
+    ) -> list:
+        """Calculates the exposure times for the fiber spectrograph and
+        the electrometer given camera exposure times and other
+        relevant configuration paramaters.
+
+        This includes wavelength dependent information for the fiber
+        spectrograph and also includes the logic for the size of
+        the electrometer buffer.
+
+        Parameters
+        ----------
+        wavelengths : `list`
+            List of all wavelengths for this exposure list
+        config_data : `dict`
+            All information from configuration file
+
+        Returns
+        -------
+        exposure_list : `array`
+            List of exposure information, including wavelength and camera,
+            fiberspectrograph and electrometer exposure times.
+
+        """
+        exposures = []
+        for wavelength in wavelengths:
+            for exptime in config_data["exposure_times"]:
+                if config_data["use_electrometer"]:
+                    time_sep = (
+                        config_data["electrometer_integration_time"] * 3.07
+                    ) + 0.00254
+                    max_exp_time = 16667 * time_sep  # buffer size
+                    if exptime > max_exp_time:
+                        electrometer_exptime = max_exp_time
+                        self.log.info(
+                            f"Electrometer exposure time reduced to {max_exp_time}"
+                        )
+                    else:
+                        electrometer_exptime = exptime
+                else:
+                    electrometer_exptime = np.nan
```

Comment:
Consider using `None` instead of `np.nan`, it has a smaller memory footprint and will make it easier to check it later.

Reply:
As mentioned above, the reason for using np.nan instead of None is that there are several places that have float(exposure_time) and np.nan will not fault. If it takes up too much memory I can refactor things, but this seemed a good way to keep most of the other code intact.